src/app/backend/client/manager.go

```diff
@@ -97,6 +100,38 @@ func (self *clientManager) Config(req *restful.Request) (*rest.Config, error) {
 	return cfg, nil
 }
 
+// ClientCmdConfig creates ClientCmd Config based on authentication information extracted from request.
+// Currently request header is only checked for existence of 'Authentication: BearerToken'
+func (self *clientManager) ClientCmdConfig(req *restful.Request) (clientcmd.ClientConfig, error) {
+	authInfo := self.extractAuthInfo(req)
+
+	cfg, err := self.buildConfigFromFlags(self.apiserverHost, self.kubeConfigPath)
+	if err != nil {
+		return nil, err
+	}
+
+	if len(authInfo.Token) > 0 {
```

Comment:
Shouldn't this check only set `cmdCfg.AuthInfos[DefaultCmdConfigName] = &authInfo` and config be returned anyway similar to what `Config` method does? Otherwise when user is not using authorization header and `apiserver-host` is used instead of in-cluster config deploy will fail because `cmdutil.NewFactory(nil)` assumes that dashboard runs using in-cluster config.